### PR TITLE
[CNXC-249] Patient MRN not displaying after search/re-routing

### PR DESCRIPTION
**Pull request by colinywu**
### Problem Statement
Currently, the patient MRN box is empty after performing a search, and also after routing back to the Create Analysis page.

### Implementation
Added createAnalysis store into CreateAnalysisDetail useEffect's dependency to now update and display the current patient MRN correctly, and also keep/store the value when re-routing back to the Create Analysis page during the same session. The value now stays in sync with Create Analysis page patient content.

### CODE REVIEW

File: src/components/PatientLookup.tsx [hunk 1]
```diff
@@ -25,14 +22,14 @@ interface PatientLookupProps {
 }
 
 const PatientLookup = (props: PatientLookupProps) => {
-  const { dispatch } = useContext(AppContext);
+  const { state: {createAnalysis: {patientID}}, dispatch } = useContext(AppContext);
```
Comment (janakitti) [inline src/components/PatientLookup.tsx:25]:
```suggestion
  const { state: { createAnalysis: { patientID } }, dispatch } = useContext(AppContext);
```

**Review by janakitti — CHANGES_REQUESTED** [review]
Just a minor code formatting change.

File: src/components/PatientLookup.tsx [hunk 2]
```diff
@@ -25,14 +22,14 @@ interface PatientLookupProps {
 }
 
 const PatientLookup = (props: PatientLookupProps) => {
-  const { dispatch } = useContext(AppContext);
+  const { state: {createAnalysis: { patientID }}, dispatch } = useContext(AppContext);
```
Comment (janakitti) [inline src/components/PatientLookup.tsx:25]:
```suggestion
  const { state: { createAnalysis: { patientID }}, dispatch } = useContext(AppContext);
```

**Review by boris-nguyen — APPROVED** [review]
Good job @colinywu !

**Review by janakitti — APPROVED** [review]
Looks great!

**Pull request merged**